### ISSUE TITLE
Support streaming results generation from the /iox/api/v1/database/:name/query endpoint

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#873]
__Use case:__
As of now the query endpoint implementation will buffer all results of running a query into `RecordBatch`es and then convert the entire thing into a single in memory string, and return that. This means:
1. There is no way to cancel a long running query if the client hangs up
2. The client doesn't get any results before all results have been processed and converted
3. Peak memory is higher than it need be

See https://github.com/influxdata/influxdb_iox/pull/870 for more details.

__Proposal:__
Use the Hyper http body and produce results one record batch at a time. 

__Current behaviour:__
See usecase

__Desired behaviour:__
Describe what you want.

__Alternatives considered:__
Describe other solutions or features you considered.

**Comment by jacobmarble** [comment]
@alamb does the query endpoint still batch the response in memory before streaming it to the client?

**Comment by alamb** [comment]
Yes, it still buffers the results (as both a record batch and then as a large string) prior to sending it to the client
